src/route/seg6.rs

```diff
@@ -0,0 +1,190 @@
+            0 => Seg6Mode::Inline,
+            1 => Seg6Mode::Encap,
+            v => Seg6Mode::Other(v),
+        }
+    }
+}
+
+const SEG6_IPTUNNEL_SRH: u16 = 1;
+
+/// Netlink attributes for `RTA_ENCAP` with `RTA_ENCAP_TYPE` set to
+/// `LWTUNNEL_ENCAP_SEG6`.
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[non_exhaustive]
+pub enum RouteSeg6IpTunnel {
+    // Use an IPv6 segment routing header
+    Seg6(Seg6Header),
+    Other(DefaultNla),
+}
+
+impl Nla for RouteSeg6IpTunnel {
+    fn value_len(&self) -> usize {
+        match self {
+            RouteSeg6IpTunnel::Seg6(v) => v.value_len(),
+            RouteSeg6IpTunnel::Other(v) => v.value_len(),
+        }
+    }
+
+    fn kind(&self) -> u16 {
+        match self {
+            RouteSeg6IpTunnel::Seg6(v) => v.kind(),
+            RouteSeg6IpTunnel::Other(v) => v.kind(),
+        }
+    }
+
+    fn emit_value(&self, buffer: &mut [u8]) {
+        match self {
+            RouteSeg6IpTunnel::Seg6(v) => v.emit_value(buffer),
+            RouteSeg6IpTunnel::Other(v) => v.emit_value(buffer),
+        }
+    }
+}
+
+/// Netlink attributes for `RTA_ENCAP` with `RTA_ENCAP_TYPE` set to
+/// `LWTUNNEL_ENCAP_SEG6`.
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[non_exhaustive]
+pub struct Seg6Header {
+    // Operation mode
+    pub mode: Seg6Mode,
+    // List of segments
+    pub segments: Vec<Ipv6Addr>,
+}
+
+impl Nla for Seg6Header {
+    fn value_len(&self) -> usize {
+        let segments = match self.mode {
+            Seg6Mode::Inline => self.segments.len() + 1,
+            Seg6Mode::Encap => self.segments.len(),
+            Seg6Mode::Other(_) => self.segments.len(),
+        };
+        12 + 16 * segments
+    }
+
+    fn kind(&self) -> u16 {
+        SEG6_IPTUNNEL_SRH
+    }
+
+    fn emit_value(&self, buffer: &mut [u8]) {
+        // Some sources for understanding the format of Seg6 in Netlink
+        // https://github.com/torvalds/linux/blob/master/include/uapi/linux/seg6.h
+        // https://github.com/iproute2/iproute2/blob/main/include/uapi/linux/seg6_iptunnel.h#L27
+        // https://github.com/iproute2/iproute2/blob/e3f9681d4a777fb2595a322b421abf0036ab1aae/ip/iproute_lwtunnel.c#L952
+
+        let mut number_segments = self.segments.len();
+        if matches!(self.mode, Seg6Mode::Inline) {
+            number_segments += 1 // last segment (::) added
+        }
+
+        let srhlen = 8 + 16 * number_segments;
+
+        // mode : 4 bytes
+        NativeEndian::write_u32(&mut buffer[..4], self.mode.into());
+        // nexthdr : 1 bytes
+        buffer[4] = 0;
+        // hdrlen : 1 bytes
+        buffer[5] = ((srhlen >> 3) - 1) as u8;
+        // type : 1 byte
+        buffer[6] = 4;
+        // segments_left : 1 byte
+        buffer[7] = (number_segments - 1) as u8;
+        // first_segment : 1 byte
+        buffer[8] = (number_segments - 1) as u8;
+        // flags : 1 byte
+        buffer[9] = 0;
+        // tag : 2 bytes
+        NativeEndian::write_u16(&mut buffer[10..12], 0);
```

Comment:
Please consider to use `buffer!()` instead of hard code the byte index here. Please take `LinkMessageBuffer` as an example.